cmd/snap/cmd_run.go

```diff
@@ -748,7 +752,72 @@ func activateXdgDocumentPortal(info *snap.Info, snapApp, hook string) error {
 
 type envForExecFunc func(extra map[string]string) []string
 
+var gdbServerWelcomeFmt = `
+Welcome to "snap run --gdbserver".
+You are right before your application is run.
+Please open a different terminal and run:
+
+gdb -ex="target remote %[1]s" -ex=continue -ex="signal SIGCONT" -ex=continue
+(gdb) continue
+
+or use your favorite gdb frontend and connect to %[1]s
```

Comment:
When connecting with a frontend do you need to execute an equivalent of the above (afaiu we do)? Should we hint about that?

Reply:
Yes, one would most likely need to execute something similar. OTOH, if you use a custom frontend, you probably know what you are doing anyway.